### ISSUE TITLE
[ENH] Neural Reconciler

### ISSUE THREAD

**Issue by Spinachboul** [sktime/sktime#9304]
@felipeangelimvieira  @fkiraly 

I recently came across a paper which introduces `NeuralReconciler`.
Theoritcially speaking, it adopts neural approaches, such as deep neural networks or encider and decoder, architectures, producing coherent forecasts end to end by training with custom loss function that enforces hierarchy constraints  during optimization

It is generally preferred to capture non-linear hierarchical structures and avoids separate forecasting , reconciliation stages.
So I find that this method fills the gaps which the traditional ones can't seem to capture and would be good addition to the reconcilers list! 

Here is the attached paper : https://dl.acm.org/doi/epdf/10.1145/3616855.3635806

> I would like your thougts on this proposal, thanks!!

**Comment by felipeangelimvieira** [comment]
Hello @Spinachboul !
I'm not sure about implementing it on sktime: it seems too specific and would require complex pytorch implementations. Maybe  Pytorch Forecasting (a project also under sktime umbrella) has a better fit? 

Among ML approaches for reconciliation, I believe that [this one](https://arxiv.org/abs/2006.02043) has a better fit since it is more agnostic to the underlying model.

cc: @phoeenniixx

**Comment by Spinachboul** [comment]
@felipeangelimvieira 
Sure! yeah it does sound like more suitable for pytorch forecasting.
Do we currently have reconcilers implemented  in pytorch forecasting?
Then this would be a good enhancement?
 @phoeenniixx  @fkiraly  @PranavBhatP

**Comment by phoeenniixx** [comment]
> Do we currently have reconcilers implemented in pytorch forecasting?

No we dont :)

> Then this would be a good enhancement?

I agree! By adding separate datamodule and base models, I think we can add neural reconcilers in `ptf`, This would be a good enhancement!
